File: test/fixtures/go-project/internal/domain/order_violation.go
package domain

import "github.com/example/test-project/internal/infrastructure/postgres"

// BAD: Domain should not depend on infrastructure
// This is a deliberate violation for testing
func GetOrderFromDB(id string) (*Order, error) {
	repo := postgres.NewOrderRepository()
	return repo.FindByID(id)
}

File: test/fixtures/ts-project/src/domain/order.violation.ts
// BAD: Domain should not depend on infrastructure
// This is a deliberate violation for testing
import { OrderRepository } from '@infrastructure/database/order-repository';
import type { Order } from './order';

export async function getOrderFromDB(id: string): Promise<Order> {
  const repo = new OrderRepository();
  return repo.findById(id);
}
